platform/BufferedSerial.h

```diff
@@ -85,7 +95,7 @@ class BufferedSerial : private SerialBase, public FileHandle {
     void tx_irq(void);
     void rx_irq(void);
 
-
+    void DCD_IRQ(void);
```

Comment:
The point is that it makes hang-up generic.

The PPP data pump and ATParser do not know whether they're attached to a serial port or a USB connection or whatever.

This allows the serial port to indicate the DCD hangup via the generic FileHandle API. Without that, the hangup could not be detected, or there would need to be a separate side-channel to the FileHandle API.